### DISCUSSION TITLE
Tokio::select! sync operation

### DISCUSSION THREAD

**Discussion by sandeepmukh** [tokio-rs/tokio#4011] [Q&A] [answered]
Hi! So my basic problem is that I have a synchronous operation that I need to have running concurrently with the async operations in loop statement. The way I currently have it structured is like this

```
loop {
    tokio::select {
        _ = stdin.next_line() => { do something}
   }
    do_sync_work();
} 
```
The main issue with this is that it introduces significant latency in performing the sync action. Is there a way I can either remove this latency (through a timeout maybe) or trick select! into thinking my sync action is async?

Thanks!

**Comment by Darksonn** [comment, marked as answer]
Generally one of the fundamental differences between async and sync is that sync operations simply cannot be cancelled, and that sync operations are [blocking][1]. The article behind that link has more background on this, and mentions some ways to offload the synchronous work to a thread pool, which may or may not work for you depending on the nature of the synchronous work.

If possible, it would be best to make your sync operation async.

[1]: https://ryhl.io/blog/async-what-is-blocking/

**Reply by sandeepmukh** [reply]
My sync actions is popping an item off a vector that is added by an async action. It feels like this should very naturally become an async action, but I cant figure out how to make it one.
